cs/src/Management/TunnelManagementClient.cs

```diff
@@ -519,6 +519,15 @@ private string UserLimitsPath
 
             request.Headers.UserAgent.Add(TunnelSdkUserAgent);
 
+            const string regKeyPath = @"Software\Policies\Microsoft\Tunnels";
+            var defaultOnError = true;
+            RegistryTools policies = new RegistryTools();
+            var keyValue = (string)policies.GetRegistryValueFromLocalMachineRoot(regKeyPath, defaultOnError);
+            if (keyValue != null)

```

Comment:
It'll always be not null. We return empty string if it's not windows or the registry key is missing.